baybe/kernels/base.py

```diff
@@ -19,21 +18,42 @@
 class Kernel(ABC, SerialMixin):
     """Abstract base class for all kernels."""
 
-    lengthscale_prior: Optional[Prior] = field(default=None, kw_only=True)
-    """An optional prior on the kernel lengthscale."""
-
     def to_gpytorch(self, *args, **kwargs):
         """Create the gpytorch representation of the kernel."""
         import gpytorch.kernels
 
+        # Due to the weird way that gpytorch does the inheritance for kernels, it is not
+        # sufficient to just check for the fields of the actual class, but also for the
+        # ones of the basic "Kernel" class and combine them.
         kernel_cls = getattr(gpytorch.kernels, self.__class__.__name__)
-        fields_dict = filter_attributes(object=self, callable_=kernel_cls.__init__)
+        base_cls = getattr(gpytorch.kernels, "Kernel")
+        kernel_fields_dict = filter_attributes(
+            object=self, callable_=kernel_cls.__init__
+        )
+        base_fields_dict = filter_attributes(object=self, callable_=base_cls.__init__)
+        fields_dict = kernel_fields_dict | base_fields_dict
```

Comment:
i think it was our expectation that you add this functionality into the utility
this seems to work so im also fine with it, but are there any reasons for/against it?

Reply:
Would have also expected to have this part of the utility. In fact, even a bit more: I think the base class should not be hard-coded but the utility should automatically traverse the inheritance structure upward. My question is rather how/where to stop. Going one level up works for this case, but I don't have an overview if there are somewhere multiple levels of inheritance involved (be it for kernels or other objects where we use/intend to use the utility)?!